packages/server-wallet/src/utilities/messaging.ts

```diff
@@ -0,0 +1,58 @@
+import {ChannelResult} from '@statechannels/client-api-schema';
+import {Payload} from '@statechannels/wallet-core';
+import _ from 'lodash';
+
+import {Notice} from '../protocols/actions';
+
+// Merges any messages to the same recipient to make
+export function mergeOutgoing(outgoing: Notice[]): Notice[] {
+  if (outgoing.length === 0) return outgoing;
+
+  const senders = outgoing.map(o => o.params.sender);
+  const differentSender = new Set(senders).size !== 1;
+  // This should never happen but it's nice to have a sanity
+  if (differentSender) {
+    throw new Error(`Trying to merge outgoing messages with multiple recipients ${senders}`);
+  }
+
+  const {sender} = outgoing[0].params;
+
+  const mergedOutgoing: Record<string, Payload> = {};
+
+  for (const notice of outgoing) {
+    const {recipient, data} = notice.params;
```

Comment:
Maybe you can assert on the type of data here so you don't have to repeatedly do it below?